bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmMonitorHandler.java

```diff
@@ -0,0 +1,293 @@
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.zm.action.ZmActions;
+import org.openhab.binding.zm.internal.config.ZmMonitorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ZmMonitorHandler} represents a Zoneminder monitor. The monitor handler
+ * interacts with the server bridge to communicate with the Zoneminder server.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class ZmMonitorHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ZmMonitorHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    private @Nullable ZmBridgeHandler bridgeHandler;
+
+    private @NonNullByDefault({}) String monitorId;
+    private @Nullable Integer imageRefreshInterval;
+    private Integer alarmDuration = DEFAULT_ALARM_DURATION_SECONDS;
+
+    private @Nullable ScheduledFuture<?> imageRefreshJob;
+    private @Nullable ScheduledFuture<?> alarmOffJob;
+
+    public ZmMonitorHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        ZmMonitorConfig config = getConfigAs(ZmMonitorConfig.class);
+        monitorId = config.monitorId;
+        imageRefreshInterval = config.imageRefreshInterval;
+        Integer value = config.alarmDuration;
+        alarmDuration = value != null ? value : DEFAULT_ALARM_DURATION_SECONDS;
+        bridgeHandler = (ZmBridgeHandler) getBridge().getHandler();
+        updateStatus(ThingStatus.ONLINE);
+        startImageRefreshJob();
+    }
+
+    @Override
+    public void dispose() {
+        stopAlarmOffJob();
+        turnAlarmOff();
+        stopImageRefreshJob();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // No need to handle as status will be updated at the next refresh interval
+            return;
+        }
+        logger.debug("Monitor {}: Received command '{}' for channel '{}'", monitorId, command, channelUID.getId());
+        ZmBridgeHandler localHandler = bridgeHandler;
+        if (localHandler == null) {
+            logger.info("Monitor {}: Can't execute command because bridge handler is null", monitorId);
```

Comment:
See above

Reply:
Implemented cache of channel states to enable handling of REFRESH.